bootloader/src/old_machine_bootloader_linux.c

```diff
@@ -0,0 +1,290 @@
+#ifndef _WIN32
+
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include <sys/utsname.h>
+#include <unistd.h>
+#include <gnu/libc-version.h>
+#include <netdb.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <ifaddrs.h>
+#include <curl/curl.h>
+
+#define minVersion 6.1
+
+#define BOOTLOADER_SERVER_PORT ":5001"
+#define ISLAND_SERVER_PORT ":5000"
+
+void error(const char *msg) { perror(msg); exit(1); }
+
+struct response {
+  char *ptr;
+  size_t len;
+};
+
+struct requestData {
+    char* IPstring;
+    char* glibcVersion;
+    char* hostname;
+    char* osVersion;
+    bool tunnelUsed;
+    char* tunnel;
+};
+
+char* getRequestDataJson(struct requestData reqData){
+    char tunnel[26];
+    if(reqData.tunnelUsed){
+        snprintf(tunnel, 26, "%s%s%s", "\"", reqData.tunnel, "\"");
+    } else {
+        strcpy(tunnel,"false");
+    }
+    // Concatenate into string for post data
+    char* buf = malloc(sizeof(char) * (500));
+    snprintf(buf,
+             sizeof(char) * (500),
+             "{\"system\":\"%s\", \"os_version\":\"%s\", \"glibc_version\":\"%s\", \"hostname\":\"%s\", \"tunnel\":%s, \"ips\": [\"%s\"]}",
+             "linux",
+             reqData.osVersion,
+             reqData.glibcVersion,
+             reqData.hostname,
+             tunnel,
+             reqData.IPstring);
+    return buf;
+}
+
```

Comment:
Nearly identical across both OSes. Refactor.

Reply:
Since it builds different json output (one with glibc_version and linux) and one returns wide char instead of char I don't think it's worth it. I think having two similar functions is more readable than having one with a bunch of if statements.

Reply:
The difference in JSON is a single if.
And there are portable ways to handle wide char but less important